src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -527,6 +555,26 @@ export class WorkspaceSavedObjectsClientWrapper {
       return await wrapperOptions.client.deleteByWorkspace(workspace, options);
     };
 
+    let isDashboardAdmin: boolean = false;
+    if (this.applicationConfig) {
+      const workspaceState = getWorkspaceState(wrapperOptions.request);
+      isDashboardAdmin = workspaceState?.isDashboardAdmin || false;
+    } else {
+      const config = this.config || ({} as WorkspacePluginConfigType);
+      const adminGroups = config.dashboardAdmin?.groups || [];
+      const adminUsers = config.dashboardAdmin?.users || [];
+      isDashboardAdmin = isRequestByDashboardAdmin(
+        wrapperOptions.request,
+        adminGroups,
+        adminUsers,
+        this.permissionControl
+      );
+    }
```

Comment:
Please move the logic to the postAuth handler. The clientWrapper should only detect if `isDashbaordAdmin` is inside the workspaceState

Reply:
But in that case, the logic to judge `isDashboardAdmin` will be split to two places. For permission check wrapper, it does not have to be aware of such logic.

Reply:
Agree with @SuZhou-Joe , the `isDashboarAdmin` flag can be shared to other places after move static config logic to postAuth. It's make sense that the final dashboard admin state stored in request.

Reply:
Yes, I have move the logic to the postAuth handler.